scripts/eslint.py

```diff
@@ -0,0 +1,75 @@
+"""  # pylint: disable=django-not-configured
+Check code quality using pycodestyle, pylint, and diff_quality.
+"""
+
+import re
+import subprocess
+import shlex
+import sys
+
+
+class BuildFailure(Exception):
+    pass
+
+
+def fail_quality(name, message):
+    """
+    Fail the specified quality check.
+    """
+
+    print(name)
+    raise BuildFailure(message)
+
+
+def run_eslint():
+    """
+    Runs eslint on static asset directories.
+    If limit option is passed, fails build if more violations than the limit are found.
+    """
+    violations_limit = 1285
+
+    command = [
+        "node",
+        "--max_old_space_size=4096",
+        "node_modules/.bin/eslint",
+        "--ext", ".js",
+        "--ext", ".jsx",
+        "--format=compact",
+        "lms",
+        "cms",
+        "common",
+        "openedx",
+        "xmodule",
+    ]
+    print("Running command:", shlex.join(command))
+    result = subprocess.run(
+        command,
+        text=True,
+        check=False,
```

Comment:
```suggestion
        check=True,
```

Using check=True ensures that the script will fail if `subprocess.run` fails. Using check=False _hides_ the errors-- in other words, check=False means "fail silently".

Failing silently is usually bad. In CI scripting, failing silently is **very bad**, because it means that PRs can pass tests even when they shouldn't.

@salman2013 , every time you call `subprocess.run`, including here, please please use check=True unless you are actively handling the error case yourself.

Reply:
@kdmccormick 
You are right, actually, i intentionally used `check=False`, Reason is that `check-True `command is failing with status code 1. I tried to figure out the reason, there could be the following three possibility 
- Command Syntax
- Eslint Configuration issue
- Files path issue

The above three issues were not the reasons i investigated. The `result.stderr `is showing nothing but the `result.stdout` is printing almost 1285 errors from different files. Which i figured out the command is sending status code 1 as a lot of errors in different files. 

<img width="1060" alt="Screenshot 2024-12-08 at 12 47 25 AM" src="https://github.com/user-attachments/assets/b6fd0a95-69c1-4103-bf8c-e51cc1815da6">

As we raise exceptions in case of violation count comparison, I thought it safe to use check=False.

Reply:
Ahh I see @salman2013 . You are right, it makes sense to `check=False` here. Just please then check the value of the exit code. [According to the eslint docs](https://eslint.org/docs/latest/use/command-line-interface#exit-codes), exit code 1 indicates that linting happened successfully but there were violations, whereas exit code 2 indicates that an unexpected error occurred (in which case we'd want to fail CI).

Reply:
@salman2013 Could you open a PR to check that the exit code is 1? I have concern that an unexpected issue could one day cause `eslint` to fail with an exit code of 2 or higher. As long as there are any digits on the last line of output, then the script would successfully "parse" those digits as the violation count. For example, an error like this like:

```
ERROR: Got 2 arguments, expected 1
```

might be parsed as "2" violations.
```

Reply:
@kdmccormick Yes you are right, i have created a PR, could you take a look into it. https://github.com/openedx/edx-platform/pull/36068